src/device-config.ts

```diff
@@ -352,23 +353,39 @@ export function bootConfigChangeRequired(
 	// Some devices require specific overlays, here we apply them
 	ensureRequiredOverlay(deviceType, targetBootConfig);
 
+	// Search for any unsupported values
+	_.each(targetBootConfig, (value, key) => {
+		if (
+			!configBackend.isSupportedConfig(key) &&
+			currentBootConfig[key] !== value
+		) {
+			const err = `Attempt to change blacklisted config value ${key}`;
+			logger.logSystemMessage(err, { error: err }, 'Apply boot config error');
+			throw new Error(err);
+		}
+	});
+
 	if (!_.isEqual(currentBootConfig, targetBootConfig)) {
-		_.each(targetBootConfig, (value, key) => {
-			// Ignore null check because we can't get here if configBackend is null
-			if (!configBackend!.isSupportedConfig(key)) {
-				if (currentBootConfig[key] !== value) {
-					const err = `Attempt to change blacklisted config value ${key}`;
-					logger.logSystemMessage(
-						err,
-						{ error: err },
-						'Apply boot config error',
-					);
-					throw new Error(err);
-				}
+		// Check if the only difference is the targetBootConfig not containing a special case
+		const SPECIAL_CASE = 'configuration'; // ODMDATA Mode for TX2 devices
```

Comment:
This is a little concerning, after we strip the namespace we're left with simply `configuration`. I could see this causing name clashes in the future, and definitely something that we should be aware of.